ambry-router/src/main/java/com.github.ambry.router/GetManager.java

```diff
@@ -123,8 +129,9 @@ void submitGetBlobOperation(String blobId, FutureResult<ReadableStreamChannel> f
     try {
       GetBlobOperation getBlobOperation =
           new GetBlobOperation(routerConfig, routerMetrics, clusterMap, responseHandler, blobId, futureResult, callback,
-              operationCompleteCallback, blobIdFactory, time);
+              operationCompleteCallback, readyForPollCallback, blobIdFactory, time);
       getOperations.add(getBlobOperation);
+      readyForPollCallback.onPollReady();
```

Comment:
AFAIK, the `OperationController` doesn't have a thread? The `OperationController` code runs in the context of the operation-submitting thread.

In fact, all I'm recommending is that the code be moved. Not the context of thread where the code is run.